core/src/main/java/org/elasticsearch/action/support/tasks/BaseTasksRequest.java

```diff
@@ -0,0 +1,197 @@
+        this.actions = actions;
+        return (T) this;
+    }
+
+    /**
+     * Return the list of action masks for the actions that should be returned
+     */
+    public String[] actions() {
+        return actions;
+    }
+
+    public final String[] nodesIds() {
+        return nodesIds;
+    }
+
+    @SuppressWarnings("unchecked")
+    public final T nodesIds(String... nodesIds) {
+        this.nodesIds = nodesIds;
+        return (T) this;
+    }
+
+    /**
+     * Returns the parent node id that tasks should be filtered by
+     */
+    public String parentNode() {
+        return parentNode;
+    }
+
+    @SuppressWarnings("unchecked")
+    public T parentNode(String parentNode) {
+        this.parentNode = parentNode;
+        return (T) this;
+    }
+
+    /**
+     * Returns the parent task id that tasks should be filtered by
+     */
+    public long parentTaskId() {
+        return parentTaskId;
+    }
+
+    @SuppressWarnings("unchecked")
+    public T parentTaskId(long parentTaskId) {
+        this.parentTaskId = parentTaskId;
+        return (T) this;
+    }
+
+
+    public TimeValue timeout() {
+        return this.timeout;
+    }
+
+    @SuppressWarnings("unchecked")
+    public final T timeout(TimeValue timeout) {
+        this.timeout = timeout;
+        return (T) this;
+    }
+
+    @SuppressWarnings("unchecked")
+    public final T timeout(String timeout) {
+        this.timeout = TimeValue.parseTimeValue(timeout, null, getClass().getSimpleName() + ".timeout");
+        return (T) this;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        nodesIds = in.readStringArray();
+        actions = in.readStringArray();
+        parentNode = in.readOptionalString();
+        parentTaskId = in.readLong();
+        if (in.readBoolean()) {
+            timeout = TimeValue.readTimeValue(in);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeStringArrayNullable(nodesIds);
+        out.writeStringArrayNullable(actions);
+        out.writeOptionalString(parentNode);
+        out.writeLong(parentTaskId);
+        out.writeOptionalStreamable(timeout);
+    }
+
+    public boolean match(Task task) {
+        if (actions() != null && actions().length > 0 && Regex.simpleMatch(actions(), task.action()) == false) {
+            return false;
+        }
+        if (parentNode() != null) {
+            if (task instanceof ChildTask == false) {
+                // not child task - cannot match parent node
+                return false;
+            }
+            if (parentNode().equals(((ChildTask) task).parentNode()) == false) {
+                return false;
+            }
+        }
+        if (parentTaskId() != BaseTasksRequest.ALL_TASKS) {
+            if (task instanceof ChildTask == false) {
+                // not child task - cannot match parent node
```

Comment:
Wrong comment I think.

Totally optional: I'd probably write this in a different way:
1. Return early if parentNode == null and parentTaskId == ALL_TASKS.
2. Return early if task instanceof ChildTask == false
3. Then do all your tests.

I think that is easier to read and only has a single instanceof check which always warms my heart.